src/plugins/saved_objects_management/public/services/types/action.ts

```diff
@@ -37,8 +38,13 @@ export abstract class SavedObjectsManagementAction {
 
   private callbacks: Function[] = [];
 
+  protected capabilities: ApplicationStart['capabilities'] | null = null;
   protected record: SavedObjectsManagementRecord | null = null;
 
+  public setUiCapabilities(capabilities: ApplicationStart['capabilities']) {
+    this.capabilities = capabilities;
+  }
```

Comment:
Do you really need that new method? Shouldn't the action implementation be able to retrieve the `capabilities` itself instead? The action is executed at runtime, so a `getStartServices` approach when registering the action would allow to do so.

Or did I miss anything here?

Reply:
The action needs to know the application capabilities before it decides whether or not it should be enabled. Unfortunately, actions are created during the plugin `setup` lifecycle phase, so they don't yet have access to capabilities. There's currently no point in the action lifecycle where it can wait for the `getStartServices` promise to resolve.

So, without introducing a race condition, the options that I can see are:
* This approach -- consumers of actions must first "hydrate" each action with the currently available capabilities, or
* Something similar to Column.loadData() ([snippet 1](https://github.com/elastic/kibana/blob/aaadbe88c5b6e5d08e2ccb1bdda753f84858d94d/src/plugins/saved_objects_management/public/services/types/column.ts#L28), [snippet 2](https://github.com/elastic/kibana/blob/aaadbe88c5b6e5d08e2ccb1bdda753f84858d94d/src/plugins/saved_objects_management/public/management_section/objects_table/components/table.tsx#L97-L104), [snippet 3](https://github.com/elastic/kibana/blob/aaadbe88c5b6e5d08e2ccb1bdda753f84858d94d/x-pack/plugins/spaces/public/share_saved_objects_to_space/share_saved_objects_to_space_column.tsx#L138-L152)). That seemed a bit unnecessary to me for this since we're really not loading any data, just waiting on start services to resolve.

I'd be happy to change it if there's a better / more desirable way to achieve this.

Reply:
> There's currently no point in the action lifecycle where it can wait for the getStartServices promise to resolve.

Fair point. That would force to load / list the actions in the table component asynchronously, which would introduce more changes and complexity than what you did.

To futur-proof this API, maybe we should have something like `setActionContext` instead of `setUiCapabilities` though.

Would be
```ts
interface ActionContext {
  capabilities: Capabilities
}
```

for now, but would avoid adding additional setters to the `Action` if we need to transfer more data to it in the future.

Would that make sense to you?

Reply:
Yes! Done in a95bcaf.